src/Channels/StreamExtensions.cs

```diff
@@ -23,6 +49,91 @@ public static class StreamExtensions
             }
         }
 
+        private class StreamUnpooledChannelAdaptor : Stream
+        {
+            private readonly Stream _stream;
+            private readonly UnpooledChannel _channel;
+            private CancellationToken cancellationToken;
+
+            public IReadableChannel Channel => _channel;
+
+            public override bool CanRead => false;
+            public override bool CanSeek => false;
+            public override bool CanWrite => true;
+
+            public override long Length
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public override long Position
+            {
+                get
+                {
+                    throw new NotImplementedException();
+                }
+
+                set
+                {
+                    throw new NotImplementedException();
+                }
+            }
+
+            public StreamUnpooledChannelAdaptor(Stream stream)
+            {
+                _stream = stream;
+                _channel = new UnpooledChannel();
+            }
+
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                WriteAsync(buffer, offset, count).Wait();
+            }
+
+            public override async Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+            {
+                await _channel.WriteAsync(new ArraySegment<byte>(buffer, offset, count), cancellationToken);
+            }
+
+            // *gasp* Async Void!? It works here because we still have _channel.Writing to track completion.
+            internal async void Produce(CancellationToken cancellationToken)
+            {
+                try
+                {
+                    // We have to provide a buffer size in order to provide a cancellation token. Weird but meh.
```

Comment:
But I suppose we can prevent any of the CopyToAsync set-up happening until a Reader arrives.